### DISCUSSION TITLE
How to let a container that needs port 80 internally through Traefik?

### DISCUSSION THREAD

**Discussion by CKyle22** [htpcBeginner/docker-traefik#274] [Q&A] [answered]
I'm trying to add [Grocy](https://hub.docker.com/r/linuxserver/grocy) to my list of services that I used this guide to set up. I have added multiple services since starting, so this isn't my first time.

You'll see in the link that Grocy uses port 80 internally and 9283 externally. But I have no idea how to set that up! Adding it my usual way just results in a 502 error when I visit the URL (although according to the Traefik dashboard it's running fine).

Here's the relevant part of my compose file:

```  # Grocy
  grocy:
    image: lscr.io/linuxserver/grocy:latest
    container_name: grocy
    networks:
      - t2_proxy
#    ports:
#      - 9283:80
    volumes: 
      - $DOCKERDIR/grocy:/config
    environment:
      - PUID=$PUID
      - PGID=$PGID
      - TZ=$TZ
    restart: unless-stopped
    labels:
      - "traefik.enable=true"
      ## HTTP Routers
      - "traefik.http.routers.grocy-rtr.entrypoints=https"
      - "traefik.http.routers.grocy-rtr.rule=Host(`grocy.$DOMAINNAME`)"
      - "traefik.http.routers.grocy-rtr.tls=true"
      ## Middlewares
      - "traefik.http.routers.grocy-rtr.middlewares=chain-oauth@file"
      ## HTTP Services
      - "traefik.http.routers.grocy-rtr.service=grocy-svc"
      - "traefik.http.services.grocy-svc.loadbalancer.server.port=9283" 
```

Every other service I've added, the internal and external ports were the same, so I don't know how to add this one to my stack of services!

**Comment by tehniemer** [comment, marked as answer]
`- "traefik.http.services.grocy-svc.loadbalancer.server.port=80"`

**Comment by CKyle22** [comment]
Thank you, had no idea that's how `loadbalancer.server.port` worked
